File: python/test_standard_calc.py
from standard_calc import bound_to_180, is_angle_between


""" Tests for bound_to_180() """


def test_bound_basic1():
    assert bound_to_180(0) == 0
    assert bound_to_180(180) == -180
    assert bound_to_180(210) == -150
    assert bound_to_180(-180) == -180
    assert bound_to_180(-210) == 150
    assert bound_to_180(750) == 30
    assert bound_to_180(-750) == -30
    print("bound basic 1 Passed")


""" Tests for is_angle_between() """


def test_between_basic1():
    assert is_angle_between(0, 1, 2)
    assert is_angle_between(120, 180, 330) == False
    assert is_angle_between(330, 180, 120) == False
    assert is_angle_between(120, 90, 330)
    assert is_angle_between(330, 90, 120)
    assert is_angle_between(90, 120, 180)
    assert is_angle_between(180, 120, 90)
    assert is_angle_between(90, 210, 180) == False
    assert is_angle_between(180, 210, 90) == False
    assert is_angle_between(270, 30, 330) == False
    assert is_angle_between(30, 30, 30)
    assert is_angle_between(-30, 0, 30)
    assert is_angle_between(30, 30, 90)
    assert is_angle_between(30, 90, 90)


test_bound_basic1()
test_between_basic1()

File: c-cpp/test_standard_calc.cpp
#include "stdio.h"
#include "cutest/CuTest.h"
#include "standard_calc.h"

/**************** Tests for bound_to_180() *******************/
void test_bound_basic1(CuTest *tc) {
    CuAssertDblEquals(tc, 0, bound_to_180(0), 0.0001);
}

/**************** Tests for is_angle_between() *******************/
void test_between_basic1(CuTest *tc) {
    CuAssertTrue(tc, is_angle_between(0, 1, 2));
}

int main(int argc, char const *argv[]) {
    CuString *output = CuStringNew();
    CuSuite *suite = CuSuiteNew();

    SUITE_ADD_TEST(suite, test_bound_basic1);
    SUITE_ADD_TEST(suite, test_between_basic1);

    CuSuiteRun(suite);
    CuSuiteSummary(suite, output);
    CuSuiteDetails(suite, output);
    printf("%s\n", output->buffer);

    return suite->failCount > 0 ? 1 : 0;
}
